src/java/org/apache/cassandra/cql3/functions/FormatFcts.java

```
@@ -0,0 +1,426 @@
+     *
+     * <pre>
+     * format_time(val, 'm', 's')
+     * </pre>
+     * <p>
+     * Values can be max of Long.MAX_VALUE, If the conversion produces overflown value, Long.MAX_VALUE will be returned.
+     * <p>
+     * Supported units are: d, h, m, s, ms, us, µs, ns
+     * <p>
+     * Supported column types on which this function is possible to be applied:
+     * <pre>INT, TINYINT, SMALLINT, BIGINT, VARINT, ASCII, TEXT</pre>
+     * For ASCII and TEXT types, text of such column has to be a non-negative number.
+     * <p>
+     * The conversion of negative values is not supported.
+     */
+    public static class FormatTimeFct extends NativeScalarFunction
+    {
+        private static final String FUNCTION_NAME = "format_time";
+
+        private FormatTimeFct(AbstractType<?>... argsTypes)
+        {
+            super(FUNCTION_NAME, UTF8Type.instance, argsTypes);
+        }
+
+        @Override
+        public ByteBuffer execute(Arguments arguments) throws InvalidRequestException
+        {
+            if (arguments.get(0) == null)
+                return null;
+
+            if (arguments.containsNulls())
+                throw new InvalidRequestException("none of the arguments may be null");
+
+            long value = getValue(arguments);
+
+            if (value < 0)
+                throw new InvalidRequestException("value must be non-negative");
+
+            if (arguments.size() == 1)
+            {
+                Pair<Long, String> convertedValue = convertValue(value);
+                return UTF8Type.instance.fromString(convertedValue.left + " " + convertedValue.right);
+            }
+
+            TimeUnit sourceUnit;
+            TimeUnit targetUnit;
+            String targetUnitAsString;
+
+            if (arguments.size() == 2)
+            {
+                sourceUnit = MILLISECONDS;
+                targetUnitAsString = arguments.get(1);
+            }
+            else
```

Should we also check for `arguments.size() > 3`? I think we should also return an `InvalidRequestException on that case instead of silently ignoring the extra args.

---

Yeah, I believe it's already handled in the ``FunctionFactory::doGetOrCreateFunction`` method and there is a test case checking that.